Upgrade Gradle to 7.4 + lib upgrades + Target SDK 31 + publish test report at Github actions

TranceLove opened the pull request:
Fixes #3224. Let the crash begin...

TranceLove committed:
Upgrade Gradle to 7.4

TranceLove committed:
Add workflow step to publish test report

TranceLove committed:
Library upgrades

- Kotlin 1.6.10
- Robolectric 4.7.3
- Room 2.4.2
- BouncyCastle 1.70
- AndroidX Core 1.7.0
- AndroidX Fragment 1.4.1
- AndroidX Annotation 1.3.0
- AndroidX Preferences 1.2.0
- Google Play Billing Client 4.1.0
- Apache ftpserver 1.1.1

Alongside required code changes while upgrading AndroidX Fragment.

Also reduced JVM heap size for tests.

TranceLove commented:
Looks like we found the murderer.

```
2022-03-19T13:48:13.6528146Z 2022-03-19T13:48:12.298+0000 [DEBUG] [TestEventLogger] com.amaze.filemanager.asynchronous.services.ExtractServiceTest > testZeroByteArchive STANDARD_ERROR
2022-03-19T13:48:13.6529510Z 2022-03-19T13:48:12.298+0000 [DEBUG] [TestEventLogger]     io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.RuntimeException: Can't toast on a thread that has not called Looper.prepare()
2022-03-19T13:48:13.6531004Z 2022-03-19T13:48:12.306+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.observers.EmptyCompletableObserver.onError(EmptyCompletableObserver.java:50)
2022-03-19T13:48:13.6531994Z 2022-03-19T13:48:12.307+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.onError(CompletableSubscribeOn.java:74)
2022-03-19T13:48:13.6533071Z 2022-03-19T13:48:12.307+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableFromRunnable.subscribeActual(CompletableFromRunnable.java:40)
2022-03-19T13:48:13.6533904Z 2022-03-19T13:48:12.308+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6534770Z 2022-03-19T13:48:12.308+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
2022-03-19T13:48:13.6535723Z 2022-03-19T13:48:12.308+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.schedulers.TrampolineScheduler.scheduleDirect(TrampolineScheduler.java:52)
2022-03-19T13:48:13.6536719Z 2022-03-19T13:48:12.308+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn.subscribeActual(CompletableSubscribeOn.java:38)
2022-03-19T13:48:13.6537721Z 2022-03-19T13:48:12.309+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6538386Z 2022-03-19T13:48:12.309+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2295)
2022-03-19T13:48:13.6539166Z 2022-03-19T13:48:12.309+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.runInApplicationThread(AppConfig.java:184)
2022-03-19T13:48:13.6539963Z 2022-03-19T13:48:12.310+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.toast(AppConfig.java:174)
2022-03-19T13:48:13.6540795Z 2022-03-19T13:48:12.310+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:329)
2022-03-19T13:48:13.6541697Z 2022-03-19T13:48:12.310+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:227)
2022-03-19T13:48:13.6542466Z 2022-03-19T13:48:12.310+0000 [DEBUG] [TestEventLogger]     	at android.os.AsyncTask$2.$$robo$$android_os_AsyncTask_2$call(AsyncTask.java:333)
2022-03-19T13:48:13.6543083Z 2022-03-19T13:48:12.311+0000 [DEBUG] [TestEventLogger]     	at android.os.AsyncTask$2.call(AsyncTask.java)
2022-03-19T13:48:13.6543706Z 2022-03-19T13:48:12.311+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-19T13:48:13.6544404Z 2022-03-19T13:48:12.311+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2022-03-19T13:48:13.6545267Z 2022-03-19T13:48:12.312+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-19T13:48:13.6546010Z 2022-03-19T13:48:12.312+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2022-03-19T13:48:13.6546804Z 2022-03-19T13:48:12.312+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2022-03-19T13:48:13.6547450Z 2022-03-19T13:48:12.313+0000 [DEBUG] [TestEventLogger]     	at java.base/java.lang.Thread.run(Thread.java:829)
2022-03-19T13:48:13.6548273Z 2022-03-19T13:48:12.313+0000 [DEBUG] [TestEventLogger]     Caused by: java.lang.RuntimeException: Can't toast on a thread that has not called Looper.prepare()
2022-03-19T13:48:13.6549104Z 2022-03-19T13:48:12.313+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast$TN.__constructor__(Toast.java:393)
2022-03-19T13:48:13.6549680Z 2022-03-19T13:48:12.314+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast$TN.<init>(Toast.java)
2022-03-19T13:48:13.6550250Z 2022-03-19T13:48:12.314+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast.__constructor__(Toast.java:117)
2022-03-19T13:48:13.6550867Z 2022-03-19T13:48:12.315+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast.<init>(Toast.java)
2022-03-19T13:48:13.6551513Z 2022-03-19T13:48:12.316+0000 [DEBUG] [TestEventLogger]     	at org.robolectric.shadows.ShadowToast.makeText(ShadowToast.java:37)
2022-03-19T13:48:13.6552131Z 2022-03-19T13:48:12.316+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast.makeText(Toast.java)
2022-03-19T13:48:13.6552807Z 2022-03-19T13:48:12.316+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.lambda$toast$1(AppConfig.java:174)
2022-03-19T13:48:13.6553745Z 2022-03-19T13:48:12.317+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableFromRunnable.subscribeActual(CompletableFromRunnable.java:36)
2022-03-19T13:48:13.6554572Z 2022-03-19T13:48:12.317+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6555413Z 2022-03-19T13:48:12.317+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
2022-03-19T13:48:13.6556375Z 2022-03-19T13:48:12.317+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.schedulers.TrampolineScheduler.scheduleDirect(TrampolineScheduler.java:52)
2022-03-19T13:48:13.6557353Z 2022-03-19T13:48:12.318+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn.subscribeActual(CompletableSubscribeOn.java:38)
2022-03-19T13:48:13.6558194Z 2022-03-19T13:48:12.323+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6558835Z 2022-03-19T13:48:12.323+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2295)
2022-03-19T13:48:13.6559589Z 2022-03-19T13:48:12.323+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.runInApplicationThread(AppConfig.java:184)
2022-03-19T13:48:13.6560383Z 2022-03-19T13:48:12.326+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.toast(AppConfig.java:174)
2022-03-19T13:48:13.6561204Z 2022-03-19T13:48:12.326+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:329)
2022-03-19T13:48:13.6562100Z 2022-03-19T13:48:12.326+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:227)
2022-03-19T13:48:13.6562805Z 2022-03-19T13:48:12.326+0000 [DEBUG] [TestEventLogger]     	at android.os.AsyncTask$2.call(AsyncTask.java:333)
2022-03-19T13:48:13.6563430Z 2022-03-19T13:48:12.326+0000 [DEBUG] [TestEventLogger]     	... 6 more
2022-03-19T13:48:13.6563859Z 2022-03-19T13:48:12.676+0000 [DEBUG] [TestEventLogger] 
2022-03-19T13:48:13.6564522Z 2022-03-19T13:48:12.676+0000 [DEBUG] [TestEventLogger] com.amaze.filemanager.asynchronous.services.ExtractServiceTest > testZeroByteArchive STANDARD_OUT
2022-03-19T13:48:13.6565179Z 2022-03-19T13:48:12.676+0000 [DEBUG] [TestEventLogger]     ErrorActivity reportError
2022-03-19T13:48:13.6566346Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.RuntimeException: Can't toast on a thread that has not called Looper.prepare()
2022-03-19T13:48:13.6567620Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.observers.EmptyCompletableObserver.onError(EmptyCompletableObserver.java:50)
2022-03-19T13:48:13.6568622Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.onError(CompletableSubscribeOn.java:74)
2022-03-19T13:48:13.6569654Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableFromRunnable.subscribeActual(CompletableFromRunnable.java:40)
2022-03-19T13:48:13.6570498Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6571344Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
2022-03-19T13:48:13.6572765Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.schedulers.TrampolineScheduler.scheduleDirect(TrampolineScheduler.java:52)
2022-03-19T13:48:13.6573923Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn.subscribeActual(CompletableSubscribeOn.java:38)
2022-03-19T13:48:13.6574864Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6575609Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2295)
2022-03-19T13:48:13.6576460Z 2022-03-19T13:48:12.677+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.runInApplicationThread(AppConfig.java:184)
2022-03-19T13:48:13.6577680Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.toast(AppConfig.java:174)
2022-03-19T13:48:13.6578519Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:329)
2022-03-19T13:48:13.6579405Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:227)
2022-03-19T13:48:13.6580186Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at android.os.AsyncTask$2.$$robo$$android_os_AsyncTask_2$call(AsyncTask.java:333)
2022-03-19T13:48:13.6580782Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at android.os.AsyncTask$2.call(AsyncTask.java)
2022-03-19T13:48:13.6581407Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-19T13:48:13.6582115Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2022-03-19T13:48:13.6582799Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-03-19T13:48:13.6583655Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2022-03-19T13:48:13.6584458Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2022-03-19T13:48:13.6585111Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     	at java.base/java.lang.Thread.run(Thread.java:829)
2022-03-19T13:48:13.6585794Z 2022-03-19T13:48:12.678+0000 [DEBUG] [TestEventLogger]     Caused by: java.lang.RuntimeException: Can't toast on a thread that has not called Looper.prepare()
2022-03-19T13:48:13.6586433Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast$TN.__constructor__(Toast.java:393)
2022-03-19T13:48:13.6587107Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast$TN.<init>(Toast.java)
2022-03-19T13:48:13.6587703Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast.__constructor__(Toast.java:117)
2022-03-19T13:48:13.6671215Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast.<init>(Toast.java)
2022-03-19T13:48:13.6672585Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at org.robolectric.shadows.ShadowToast.makeText(ShadowToast.java:37)
2022-03-19T13:48:13.6673354Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at android.widget.Toast.makeText(Toast.java)
2022-03-19T13:48:13.6674251Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.lambda$toast$1(AppConfig.java:174)
2022-03-19T13:48:13.6675299Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableFromRunnable.subscribeActual(CompletableFromRunnable.java:36)
2022-03-19T13:48:13.6676237Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6677275Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
2022-03-19T13:48:13.6678307Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.schedulers.TrampolineScheduler.scheduleDirect(TrampolineScheduler.java:52)
2022-03-19T13:48:13.6679357Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.internal.operators.completable.CompletableSubscribeOn.subscribeActual(CompletableSubscribeOn.java:38)
2022-03-19T13:48:13.6680253Z 2022-03-19T13:48:12.679+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2309)
2022-03-19T13:48:13.6680936Z 2022-03-19T13:48:12.684+0000 [DEBUG] [TestEventLogger]     	at io.reactivex.Completable.subscribe(Completable.java:2295)
2022-03-19T13:48:13.6681752Z 2022-03-19T13:48:12.684+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.runInApplicationThread(AppConfig.java:184)
2022-03-19T13:48:13.6682616Z 2022-03-19T13:48:12.684+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.application.AppConfig.toast(AppConfig.java:174)
2022-03-19T13:48:13.6683495Z 2022-03-19T13:48:12.684+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:329)
2022-03-19T13:48:13.6684461Z 2022-03-19T13:48:12.684+0000 [DEBUG] [TestEventLogger]     	at com.amaze.filemanager.asynchronous.services.ExtractService$DoWork.doInBackground(ExtractService.java:227)
2022-03-19T13:48:13.6702210Z 2022-03-19T13:48:12.684+0000 [DEBUG] [TestEventLogger]     	at android.os.AsyncTask$2.call(AsyncTask.java:333)
2022-03-19T13:48:13.6702981Z 2022-03-19T13:48:12.685+0000 [DEBUG] [TestEventLogger]     	... 6 more
2022-03-19T13:48:13.6703474Z 2022-03-19T13:48:12.685+0000 [DEBUG] [TestEventLogger] 
2022-03-19T13:48:13.6704632Z 2022-03-19T13:48:13.481+0000 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: FAILED
2022-03-19T13:48:13.6705531Z 2022-03-19T13:48:13.481+0000 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Process 'Gradle Test Executor 5' finished with exit value 10 (state: FAILED)
```

TranceLove committed:
Fix build.gradle after rebase with current 3.8.x codebase

TranceLove committed:
[WIP] Added quirks in RarExtractor and ZipExtractor on output path checking
For Robolectric tests successfully run on OSX.

TranceLove committed:
Gradle plugin 7.1.3

TranceLove committed:
DecryptServiceTest tweaks to wait for NotificationManager have notification

Also try use back JDK Temurin

TranceLove committed:
Upgrade Robolectric to 4.9, fix tests

Bladfool commented:
> Fixes #3224. Let the crash begin...

TranceLove committed:
Move to NotificationManagerCompat and fix failing test

TranceLove committed:
Move common test class out to separate directory

For sharing common classes between modules

TranceLove committed:
Update Gradle build plugin to 7.3.1 and target SDK version to 31

TranceLove committed:
Continue fix test

EmmanuelMess commented on app/src/test/java/com/amaze/filemanager/asynchronous/services/DecryptServiceTest.kt:
```diff
@@ -132,13 +130,26 @@ class DecryptServiceTest {
             putExtra(TAG_DECRYPT_PATH, Environment.getExternalStorageDirectory().absolutePath)
             assertEquals(START_NOT_STICKY, service.onStartCommand(this, 0, 0))
         }
```
What happened here?

VishalNehra commented on app/src/main/AndroidManifest.xml:
```diff
@@ -130,6 +131,7 @@
         </activity>
 
         <activity
+            android:exported="false"
```
Settings can be exported true no?

VishalNehra commented on app/src/main/java/com/amaze/filemanager/database/ExplorerDatabase.kt:
```diff
@@ -280,7 +280,21 @@ abstract class ExplorerDatabase : RoomDatabase() {
+                        COLUMN_CLOUD_SERVICE +
+                        " = " +
+                        COLUMN_CLOUD_SERVICE +
+                        "+1"
```
Is this solution still valid or will this give index out of bounds exception?

TranceLove replied:
To be fixed separately at #3629.

VishalNehra commented:
Let's target this with 4.x or next major release.

VishalNehra commented:
Tried publishing new version, seem like we need to target 31 now on play store console otherwise it won't allow to publish new version.

TranceLove commented:
@VishalNehra Perhaps split the SDK update part to separate PR first?

VishalNehra commented:
We need sdk update to 31. Which is most of the code in this PR. It's fine I can merge this if you can fix the conflicts.

VishalNehra commented:
We need sdk update to 31. Which is most of the code in this PR. It's fine I can merge this if you can fix the conflicts.

TranceLove committed:
Export PreferencesActivity too

VishalNehra committed:
Merge branch 'hotfix/3.8.3' of ssh://github.com/TeamAmaze/AmazeFileManager into feature/gradle-7.4-lib-upgrades

VishalNehra commented:
Notification creation is failing with following error

## Issue explanation (write below this line)



## Exception
* __App Name:__ Amaze File Manager
* __Package:__ com.amaze.filemanager.debug
* __Version:__ 3.8.3
* __User Action:__ UI Error
* __Request:__ Application crash
* __OS:__ Linux OnePlus/OnePlus9R_IND/OnePlus9R:12/SKQ1.210216.001/R.202203042255:user/release-keys 12 - 31
* __Device:__ OnePlus9R
* __Model:__ LE2101
* __Product:__ OnePlus9R_IND
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to start service com.amaze.filemanager.asynchronous.services.ZipService@f19600b with Intent { cmp=com.amaze.filemanager.debug/com.amaze.filemanager.asynchronous.services.ZipService (has extras) }: java.lang.IllegalArgumentException: com.amaze.filemanager.debug: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4825)
	at android.app.ActivityThread.access$2100(ActivityThread.java:254)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2222)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:233)
	at android.os.Looper.loop(Looper.java:344)
	at android.app.ActivityThread.main(ActivityThread.java:8212)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
Caused by: java.lang.IllegalArgumentException: com.amaze.filemanager.debug: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
	at android.app.PendingIntent.checkFlags(PendingIntent.java:378)
	at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:461)
	at android.app.PendingIntent.getActivity(PendingIntent.java:447)
	at android.app.PendingIntent.getActivity(PendingIntent.java:411)
	at com.amaze.filemanager.asynchronous.services.ZipService.onStartCommand(ZipService.kt:103)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4807)
	... 9 more

```
</details>
<hr>

TranceLove committed:
Fix PendingIntent requirement post SDK 31

As required by Android. Defaults to FLAG_IMMUTABLE per suggested in API docs.

Tested on Fairphone 3 running LineageOS 19.1 (SDK 32) and Nexus 4 running Android 4.4 (SDK 19).

VishalNehra committed:
Merge branch 'feature/gradle-7.4-lib-upgrades' of ssh://github.com/TeamAmaze/AmazeFileManager into feature/gradle-7.4-lib-upgrades

VishalNehra committed:
Run spotlessapply
